File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let clothesMap = new Map() // type별로 정리
    let res=1;
    clothes.map(([cloth, type]) => {
        if(clothesMap.has(type)){
            clothesMap.set(type, [...clothesMap.get(type), cloth])
        }
        else clothesMap.set(type, [cloth])   
})
    clothesMap.forEach((e)=>{
        res*=e.length+1
    })
    return res-1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    closet = {}
    num=[]
    answer=1
    
    for cloth, category in clothes:
        if category in closet:
            closet[category].append(cloth)
        else:
            closet[category] = [cloth]
            
    for key in closet.keys():
        num.append(len(closet[key]))

    for i in num:
        answer *= i+1
    return answer-1
        
        